Store timezone offset when starting a recording

dennisguse opened the pull request:
This PR adds a time offset to the tracks; all times remain stored as Instant (i.e., UTC).

Data migration: existing Track will use current time offset of the phone.
Import: use the time offset from the KML/GPX

Note: time offset is stored in seconds and denotes the difference from UTC.

Note: PR is based upon #1033.

Fixes #301.

* [x] add offset column to Track table
* [x] add field to Track
* [x] add check in ExportImportTest (export/import for GPX and KML)
* [ ] use offset for UI renderings: StatisticsRecordedFragment
* [ ] use offset for naming tracks (default name option)
* [ ] use offset for UI renderings: TrackListActivity
* [ ] use offset for UI renderings: MarkerListActivity
* [x] rebase onto main after merging #1033

dennisguse commented:
@iointerrupt would have time to test this?
PS: only the internal data handling is implemented, so you need to record and export for now.
Using the time offset in the UI is still to be done.

rgmf commented:
@dennisguse in AggregatedStatisticsActivity we are filtering Tracks by datetime (using phone Locale Zone). Shall I review this to adapt to this PR? I mean, TrackSelection should handle starttime_offset, right?

dennisguse commented:
@rgmf Valid point; the UI is not yet adjusted, so the data is stored, but the UI doesn't use it.
For now, this is just used for export and imports.
If you have time for this that would be awesome.
I updated the issue and added this as an open point.

rgmf commented:
@dennisguse okay, I see you are asigned to #301. I have some time so if you don't mind or you want to do other things then change the assignation and I do the list of things you have written out here: https://github.com/OpenTracksApp/OpenTracks/issues/301#issuecomment-1002138664

dennisguse commented:
@rgmf feel free to :)
I guess, I will try to refactor the TrackRecordingService in the next (free) days.
